### PR TITLE
Delayed diffs

### LINKED ISSUE #23: Asynchronous loading of diff files
Currently, scrolling through the graph may lag for commits with lots of changed files.

Loading of diffs could be done asynchronous for lag-free scrolling, while it may take a moment to show the file list.

### PR DESCRIPTION
**Pull request by mlange-42**
Fixes #23.

### CODE REVIEW

**Comment by adamreichold** [comment]
I would suggest rewording the first commit message as the event loop was not removed (it was always the `loop` on line 3744/368), only the separate event pumping thread was optimized out.

File: src/main.rs [hunk 1]
```diff
@@ -343,34 +344,42 @@ fn run(
...
-        move || loop {
-            if event::poll(update_tick_rate).unwrap() {
+        move |next_reload: Instant, next_diff: Option<Instant>| loop {
+            let deadline = if let Some(next) = next_diff {
+                next.min(next_reload)
+                    .saturating_duration_since(Instant::now())
+            } else {
+                next_reload.saturating_duration_since(Instant::now())
+            };
+            if event::poll(deadline).unwrap() {
```
Comment (adamreichold) [inline src/main.rs]: Isn't this a timeout (a duration) rather than a deadline (an instant)?

File: src/main.rs [hunk 2]
```diff
@@ -343,34 +344,42 @@ fn run(
...
                 continue;
             }
-
             return Event::Update;
         }
     };
 
     terminal.clear()?;
 
+    let mut next_diff_update = None;
```
Comment (adamreichold) [inline src/main.rs]: Since `next_event` is closure, you could move these up before its definition and access them by capturing references instead of passing them as arguments. (This would require putting them inside `Cell<_>` but this should not be an issue for `Instant`.)

**Comment by mlange-42** [comment]
> I would suggest rewording the first commit message as the event loop was not removed (it was always the `loop` on line 3744/368), only the separate event pumping thread was optimized out.

Haven't seen your comment and now there is another commit "on top of it"...

**Comment by adamreichold** [comment]
> Haven't seen your comment and now there is another commit "on top of it"...

No problem there: Do `git rebase -i master`, replace `pick` by `reword`, then save and exit, finally reword commit message, then save and exit.

File: src/main.rs (hunk 2 above)
Comment (mlange-42) [reply src/main.rs]: Mh, problems with the borrow checker when trying this. I probably should use my first `Cell` with a simpler example.

Comment (mlange-42) [reply src/main.rs]: Done.
